librz/util/print.c

```diff
@@ -221,28 +235,40 @@ RZ_API void rz_print_addr(RzPrint *p, ut64 addr) {
 				}
 			}
 			if (dec) {
-				printfmt("%s%s%" PFMT64d "%s%c", pre, white, addr, fin, ch);
+				rz_strbuf_appendf(sb, "%s%s%" PFMT64d "%s%c", pre, white, addr, fin, ch);
 			} else {
 				if (p && p->wide_offsets) {
 					// TODO: make %016 depend on asm.bits
-					printfmt("%s0x%016" PFMT64x "%s%c", pre, addr, fin, ch);
+					rz_strbuf_appendf(sb, "%s0x%016" PFMT64x "%s%c", pre, addr, fin, ch);
 				} else {
-					printfmt("%s0x%08" PFMT64x "%s%c", pre, addr, fin, ch);
+					rz_strbuf_appendf(sb, "%s0x%08" PFMT64x "%s%c", pre, addr, fin, ch);
 				}
 			}
 		} else {
 			if (dec) {
-				printfmt("%s%" PFMT64d "%c", white, addr, ch);
+				rz_strbuf_appendf(sb, "%s%" PFMT64d "%c", white, addr, ch);
 			} else {
 				if (p && p->wide_offsets) {
 					// TODO: make %016 depend on asm.bits
-					printfmt("0x%016" PFMT64x "%c", addr, ch);
+					rz_strbuf_appendf(sb, "0x%016" PFMT64x "%c", addr, ch);
 				} else {
-					printfmt("0x%08" PFMT64x "%c", addr, ch);
+					rz_strbuf_appendf(sb, "0x%08" PFMT64x "%c", addr, ch);
 				}
 			}
 		}
 	}
+	return rz_strbuf_drain(sb);
+}
+
+/// \see rz_print_addr_str
+RZ_API void rz_print_addr(RZ_NONNULL RzPrint *p, ut64 addr) {
+	rz_return_if_fail(p);
+	char *string = rz_print_addr_str(p, addr);
+	if (!string) {
+		return;
+	}
+	p->cb_printf("%s", string);
```

Comment:
before it was something like:
```c
PrintfCallback printfmt = (PrintfCallback)(p ? p->cb_printf : libc_printf);
```
so this is changing a bit how the function worked.